sip-servlets-impl/src/main/java/org/mobicents/servlet/sip/address/ParameterableImpl.java

```diff
@@ -151,16 +157,117 @@ public void setParameter(String name, String value) {
 		if(name.equalsIgnoreCase("branch") && isModifiable == ModifiableRule.Via) {
 			throw new IllegalStateException("it is forbidden to set the branch parameter on the Via Header");
 		}
-		//Fix from abondar for Issue 494 and angelo.marletta for Issue 502      
+		//Fix from abondar for Issue 494 and angelo.marletta for Issue 502	  
 		this.parameters.put(name.toLowerCase(), value);
 		if(header != null) {
 			try {
-				header.setParameter(name, "".equals(value) ? null : value);
+				if (isQuotableParameter(name, value)){
+					header.setQuotedParameter(name, value);
+				}
+				else{
+					header.setParameter(name, "".equals(value) ? null : value);
+				}
 			} catch (ParseException e) {
 				throw new IllegalArgumentException("Problem setting parameter",e);
 			}
 		}
 	}
+	
+	private static final String QUOTABLE_PARAMETER_NAME ="org.restcomm.servlets.sip.QUOTABLE_PARAMETER_NAME";
+	/**
+	 * Verify that the parameter is in the known list of parameters that its value need to be quoted
+	 * 
+	 * @return 
+	 * true: if the Parameter name is present at servletContext and its value can be quoted, else return false
+	 * otherwise
+	 */
+	protected boolean isQuotableParameter(String name, String value){
+		if (sipSession != null && sipSession.getServletContext() != null){
+			// get list of parameter names which are splited by comma.
+			String quotableParamNames = sipSession.getServletContext().getInitParameter(QUOTABLE_PARAMETER_NAME);
+			if (quotableParamNames != null){
+				String[] paramNames = quotableParamNames.split(",");
+				for (int i = 0; i < paramNames.length; i++){
+					if (name.equalsIgnoreCase(paramNames[i])){
+						if(isQuotableString(value)){

```

Comment:
even is the param is in the known list, we still deliver a rule-based logc to confirm this param must be quoted or not. This is a bit confusing, i thought we have agreed to use a predefined list of quotable params, just to prevent this kind of logic. If the logic is generic enough, then why do we need the context param. This hybrib solution its a bit confusing to me. Could you elaborate on this?